source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -120,59 +120,66 @@ bool MaterialXNode::getInternalValue(const MPlug& plug, MDataHandle& dataHandle)
     return true;
 }
 
-bool MaterialXNode::setInternalValue(const MPlug& plug, const MDataHandle& dataHandle)
+void MaterialXNode::createAndRegisterFragment(bool reloadDocument)
 {
-    auto createAndRegisterFragment = [this]()
+    try
     {
-        try
+        if (_documentFilePath.length() == 0)
+        {
+            return;
+        }
+
+        if (!_document || reloadDocument)
         {
-            if (_documentFilePath.length() == 0)
-            {
-                return;
-            }
-
-            if (!_document)
-            {
-                _document = MaterialXMaya::loadDocument(
-                    _documentFilePath.asChar(), Plugin::instance().getLibrarySearchPath()
-                );
-            };
-
-            if (_elementPath.length() == 0)
-            {
-                // When an empty element path is passed to MaterialXData's
-                // constructor, the first renderable element is selected, which is
-                // a handy feature when creating the node with the command.
-                // However this automatic behavior would complicate state
-                // transitions on attribute edits after the node has been created.
-                // So if the element path attribute is empty, bail early and don't
-                // attempt to create a MaterialXData.
-                //
-                return;
-            }
-
-            _materialXData.reset(new MaterialXData(
-                _document,
-                _elementPath.asChar(),
-                Plugin::instance().getLibrarySearchPath()
-            ));
-
-            MaterialXMaya::registerFragment(
-                _materialXData->getFragmentName(), _materialXData->getFragmentSource()
+            _document = MaterialXMaya::loadDocument(
+                _documentFilePath.asChar(), Plugin::instance().getLibrarySearchPath()
             );
+        };
+
+        if (_elementPath.length() == 0)
+        {
+            // When an empty element path is passed to MaterialXData's
+            // constructor, the first renderable element is selected, which is
+            // a handy feature when creating the node with the command.
+            // However this automatic behavior would complicate state
+            // transitions on attribute edits after the node has been created.
+            // So if the element path attribute is empty, bail early and don't
+            // attempt to create a MaterialXData.
+            //
+            return;
         }
-        catch (std::exception& e)
+
+        _materialXData.reset(new MaterialXData(
+            _document,
+            _elementPath.asChar(),
+            Plugin::instance().getLibrarySearchPath()
+        ));
+
+        MaterialXMaya::registerFragment(
+            _materialXData->getFragmentName(), _materialXData->getFragmentSource()
+        );
+
+        if (reloadDocument)
```

Comment:
Please move this command outside of this method, it can just be called afterwards.